apiserver/facades/client/application/application.go

```diff
@@ -579,7 +577,8 @@ func (api *APIBase) deployApplication(
 	if err != nil {
 		return errors.Trace(err)
 	}
-	_, err = api.deployApplicationFunc(ctx, api.backend, api.model, api.cloudService, api.credentialService, api.applicationService, api.store, DeployApplicationParams{
+	// TODO: replace model with model info/config services
```

Comment:
What is left todo here? Just for the sake of the PR and not the comment.

Reply:
just replacing the usages of state model inside `stateenvirons.SupportedFeatures`, as mentioned in my other comment. I think that is out of scope for this PR though.